oauth2_http/java/com/google/auth/oauth2/GoogleCredentials.java

```diff
@@ -166,6 +172,22 @@ public static GoogleCredentials fromStream(
             fileType, USER_FILE_TYPE, SERVICE_ACCOUNT_FILE_TYPE));
   }
 
+  /**
+   * Adds quota project ID to requestMetadata if present.
+   *
+   * @return a new map with quotaProjectId added if needed
+   */
+  static Map<String, List<String>> addQuotaProjectIdToRequestMetadata(
+      String quotaProjectId, Map<String, List<String>> requestMetadata) {
+    Preconditions.checkNotNull(requestMetadata, "requestMetadata");
```

Comment:
nit: this exception message is not particularly useful. Let's make it better or remove the message.